overlord/snapstate/snapstate.go

```diff
@@ -1313,7 +1313,9 @@ func Update(st *state.State, name, channel string, revision snap.Revision, userI
 	}
 	flat := state.NewTaskSet()
 	for _, ts := range tts {
-		flat.AddAll(ts)
+		if err := flat.AddAllWithEdges(ts); err != nil {
```

Comment:
We probably want a comment here explaining why we do this.

Relatedly we should add an explicit comment to Install and Update doc comments about the fact the task set they return has a DownloadAndChecksDoneEdge.

Reply:
was this bit:
> Relatedly we should add an explicit comment to Install and Update doc comments about the fact the task set they return has a DownloadAndChecksDoneEdge.

addressed ?

Reply:
Indeed, sorry, it is now addressed.